src/gobprepare/prepare_client.py

```diff
@@ -254,6 +261,72 @@ def action_check_row_counts(self, action: RowCountConfig) -> bool:
                 )
         return tables_counts_ok
 
+    def action_check_source_sync_complete(self, action: syncSchemaConfig) -> bool:
+        """Check import action. Verify when sync tables from databricks is completed.
+
+        :param action:
+        :return:
+        """
+
+        public_schema = action["public_schema"]
+        table_name = action["table_name"]
+        schema = action["schema"]
+        if not self._dst_datastore.query(check_table_existence_query(public_schema, table_name)):
+            raise GOBException(f"Table '{public_schema}.{table_name}' does not exists.")
+        else:
+            try:
+                where_conditions = [{ "sync_schema": schema }]
+                select_query = create_select_where_query(table_name, where_conditions)
+                record = next(self._dst_datastore.query(select_query))
+            except StopIteration:
+                raise GOBException(f"No record for schema '{schema}' found in '{public_schema}.{table_name}'.")
+            else:
+                self._execute_update_sync_schema(schema, table_name, record, update_action = 'start_prepare')
+
+    def action_complete_prepare(self, action: syncSchemaConfig) -> None:
+        """update sync table after prepare is completed.
+
+        :param action:
+        :return:
+        """
+        public_schema = action["public_schema"]
+        table_name = action["table_name"]
+        schema = action["schema"]
+        try:
+            where_conditions = [{ "sync_schema": schema }]
+            select_query = create_select_where_query(table_name, where_conditions)
```

Comment:
This function call is too generic. We don't need this because we only use this with a filtering on sync_schema. The only thing that happens with a generic query like this is that the code becomes harder to understand.

I would just create a method "get_last_schema_sync" and extract that into a method in this class, not even in the postgres utils class, as this is a locally used query. You could also move the exception catching/handling into that method, so you don't have to repeat that bit.

Main point in this is that "create_select_where_query" introduces unnecessary complexity. It's a simple query in itself.

Reply:
Hulp functions removed en simple query in de client class added